flink-core/src/main/java/org/apache/flink/core/fs/FileSystemKind.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.core.fs;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * An enumeration defining the kind and characteristics of a {@link FileSystem}.
+ * 
+ * <p>Note that the categorization here makes statements only about <i>consistency</i> and
+ * <i>directory handling</i>. It explicitly does not look at the ability to modify files,
+ * which we assume for no operation going through Flink's File System abstraction, at the moment.
+ * This might change in the future. 
+ */
+@PublicEvolving
+public enum FileSystemKind {
+
+	/**
+	 * A POSIX compliant file system, as for example found on UNIX / Linux.
+	 * 
+	 * <p>Posix file systems support directories, a consistent view, atomic renames,
+	 * and deletion of open files.
+	 */
+	POSIX_COMPLIANT(ConsistencyLevel.POSIX_STYLE_CONSISTENCY),
+
+	/**
+	 * A file system that gives a consistent view of its contents.
+	 * 
+	 * <p>File systems in this category are for example Windows file systems,
+	 * HDFS, or MapR FS. They support directories, a consistent view, but not
+	 * necessarily atomic file renaming, or deletion of open files.
+	 */
+	CONSISTENT_FILESYSTEM(ConsistencyLevel.CONSISTENT_LIST_RENAME_DELETE),
+
+	/**
+	 * A consistent object store (not an actual file system).
+	 *
+	 * <p>"File systems" of this kind support no real directories, but  and no consistent
+	 * renaming and delete operations.
+	 */
+	CONSISTENT_OBJECT_STORE(ConsistencyLevel.CONSISTENT_RENAME_DELETE),
```

Comment:
This isn't used anywhere, do we have it for future filesystems? (same applies to CONSISTENT_RENAME_DELETE)